lib/Logger.d.ts

```diff
@@ -0,0 +1,65 @@
+declare type Parameters = string | Record<string, unknown>;
+declare type ServerLoggingCallback = (logger: Logger, options: Record<string, unknown>) => Promise<void>;
```

Comment:
we can do better with the type for `options`

the only time this callback is called is [this line](https://github.com/Expensify/expensify-common/blob/f9985d572c772757885ff59ea5c82b1050c6b3f7/lib/Logger.jsx#L37).

```
const promise = this.serverLoggingCallback(this, {api_setCookie: false, logPacket: JSON.stringify(linesToLog)});
```
so we can maybe type this as `options: { api_setCookie: false, logPacket: string }`?

And for the return value, the promise is apparently resolved as an object with `requestID` property in [this line](https://github.com/Expensify/expensify-common/blob/f9985d572c772757885ff59ea5c82b1050c6b3f7/lib/Logger.jsx#L41)

it's also possible that [the return value is not actually a promise but `undefined`.
](https://github.com/Expensify/expensify-common/blob/f9985d572c772757885ff59ea5c82b1050c6b3f7/lib/Logger.jsx#L38)

```
const promise = this.serverLoggingCallback(this, {api_setCookie: false, logPacket: JSON.stringify(linesToLog)});
        if (!promise) {
            return;
        }
```

so instead of `Promise<void>` we should do `Promise<{ requestID: string }> | undefined`

Reply:
Agree, typed as suggested 👌